python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -529,6 +554,12 @@ def do_evaluate(
                 plc.concatenate.concatenate(list(tables)),
                 colnames[0],
             )
+            if include_file_paths is not None:
+                (file_paths,) = plc.filling.repeat(
+                    plc.Table([plc.interop.from_arrow(pa.array(seen_paths))]),
+                    plc.interop.from_arrow(pa.array(t.num_rows() for t in tables)),
+                ).columns()
+                df = df.with_columns([Column(file_paths, name=include_file_paths)])
```

Comment:
Any reason not to use the helper function you defined?

Reply:
No I missed that, thanks!